cli/cmd/inject_test.go

```diff
@@ -33,4 +37,29 @@ func TestInjectYAML(t *testing.T) {
 		expectedOutput := string(goldenFileBytes)
 		diffCompare(t, actualOutput, expectedOutput)
 	})
+
+	t.Run("Do not print invalid YAML on inject error", func(t *testing.T) {
+		rootTestCmd := &cobra.Command{Use: "root", Args: cobra.NoArgs, Run: emptyRun}
+
+		rootTestCmd.AddCommand(injectCmd)
+
+		buf := new(bytes.Buffer)
+		rootTestCmd.SetOutput(buf)
+		rootTestCmd.SetArgs([]string{"inject", "testdata/inject_gettest_deployment.input"})
+
+		_, err := rootTestCmd.ExecuteC()
+		if err == nil {
+			t.Fatal("Command returned nil but should have returned error")
+		}
+
+		goldenFileBytes, err := ioutil.ReadFile("testdata/inject_gettest_deployment.golden")
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		actualOutput := buf.String()
+
+		expectedOutput := string(goldenFileBytes)
+		diffCompare(t, strings.TrimSpace(actualOutput), strings.TrimSpace(expectedOutput))
```

Comment:
i can see that removing these `TrimSpace` calls crashes `test_helper.go`/`go-diff`. perhaps we have two separate issues?
1) why is the expected output different from the actual? is this a line-ending problem?
2) why does our test crash if the outputs are different?

Reply:
> why does our test crash if the outputs are different?

AFAICT, the diff stuff has always been broken. I forgot to file an issue about it but it has been the case every time I tried it.

Reply:
I've filed #390 to track this item 2.

Re: item 1, the expected output contains a single trailing newline, the actual output contains two trailing newlines. I think we should change either the expected or actual in this PR.

Reply:
@siggy I think it might be, I am not entirely sure. Trying to compare the two outputs in my editor is a little challenging since vscode seems to be removing newline characters. Do you think there is a better way to solve it than using `TrimSpace`?

Reply:
for item 2, the fix is a change in `diffCompare`:
```
-               patches := dmp.PatchMake(expected, diffs)
+               patches := dmp.PatchMake(actual, diffs)
```